### PR TITLE
Simplify must_gather_shell, get all args from user

### PR DESCRIPTION
**Pull request by myakove**
## Release Notes

- **New Features**
	- Enhanced command handling with optional namespace argument and improved error messaging.
	- Introduced new Kubernetes resources including DeploymentConfigList, DaemonSet, and various others for better resource management.
	- Simplified command syntax by removing the `-k` flag for resource kind in commands.

- **Bug Fixes**
	- Improved error handling and messaging for command execution failures.

- **Documentation**
	- Updated README to reflect new command syntax and examples.

- **Tests**
	- Added unit tests for command execution and resource retrieval to ensure reliability. 

- **Configuration**
	- Updated pre-commit configuration to include additional type stubs for better type checking.

### CODE REVIEW

**Comment by redhat-qe-bot2** [comment]
Report bugs in [Issues](https://github.com/myakove/github-webhook-server/issues)

The following are automatically added:
 * Add reviewers from OWNER file (in the root of the repository) under reviewers section.
 * Set PR size label.
 * New issue is created for the PR. (Closed when PR is merged/closed)
 * Run [pre-commit](https://pre-commit.ci/) if `.pre-commit-config.yaml` exists in the repo.

Available user actions:
 * To mark PR as WIP comment `/wip` to the PR, To remove it from the PR comment `/wip cancel` to the PR.
 * To block merging of PR comment `/hold`, To un-block merging of PR comment `/hold cancel`.
 * To mark PR as verified comment `/verified` to the PR, to un-verify comment `/verified cancel` to the PR.
        verified label removed on each new commit push.
 * To cherry pick a merged PR comment `/cherry-pick <target branch to cherry-pick to>` in the PR.
    * Multiple target branches can be cherry-picked, separated by spaces. (`/cherry-pick branch1 branch2`)
    * Cherry-pick will be started when PR is merged
 * To build and push container image command `/build-and-push-container` in the PR (tag will be the PR number).
    * You can add extra args to the Podman build command
        * Example: `/build-and-push-container --build-arg OPENSHIFT_PYTHON_WRAPPER_COMMIT=<commit_hash>`
 * To add a label by comment use `/<label name>`, to remove, use `/<label name> cancel`
 * To assign reviewers based on OWNERS file use `/assign-reviewers`
 * To check if PR can be merged use `/check-can-merge`

<details>
<summary>Supported /retest check runs</summary>

 * `/retest tox`: Retest tox
 * `/retest python-module-install`: Retest python-module-install
 * `/retest all`: Retest all

</details>

<details>
<summary>Supported labels</summary>

 * hold
 * verified
 * wip
 * lgtm

</details>

**Comment by myakove** [comment]
@jpeimer please verify (I verified but don't trust me)

File: must_gather_explorer/utils.py [hunk 1]
```diff
@@ -268,69 +273,33 @@ def get_resource_kind_by_alias(resources_aliases: dict[str, list[str]], requeste
...
     actions_dict: dict[str, Any],
     resources_aliases: dict[str, list[str]],
     all_resources: dict[str, Any],
+    action: str,
+    kind: str,
+    namespace: str = "",
+    resource_name: str = "",
+    oyaml: bool = False,
+    yaml_fields: str = "",
 ) -> bool:
```
Comment (jpeimer) [inline must_gather_explorer/utils.py:285]: So this function is not parsing actions anymore, maybe we could name it something like `call_action_function`?

**Review by jpeimer — CHANGES_REQUESTED** [review]

File: must_gather_explorer/prompt_handler.py [hunk 2]
```diff
@@ -58,6 +63,13 @@ def do_get(self, args: argparse.Namespace) -> None:
             Usage: get <resource_name>
             suported flags: -n <namespace>, -oyaml
         """
+        yaml_fields = args.yaml_fields
+        oyaml = args.oyaml
+        if yaml_fields:
+            if not oyaml:
```
[the remark this answers is not in the document]
Comment (myakove) [reply must_gather_explorer/prompt_handler.py:69]: done

File: must_gather_explorer/utils.py [hunk 3]
```diff
@@ -268,69 +273,33 @@ def get_resource_kind_by_alias(resources_aliases: dict[str, list[str]], requeste
...
+            namespace=namespace,
         )
         if not resources_raw_data:
-            CONSOLE.print(f"No resources found for {resource_kind} {resource_name} {namespace_name}")
+            CONSOLE.print(f"No resources found for {kind} {resource_name} {namespace}")
             return False
-        actions_dict[action_name](resources_raw_data, print_yaml, yaml_fields_to_get)
+        actions_dict[action](resources_raw_data, oyaml, yaml_fields)
 
     except MissingResourceKindAliasError:
```
[the remark this answers is not in the document]
Comment (myakove) [reply must_gather_explorer/utils.py:304]: Done

File: must_gather_explorer/utils.py (hunk 1 above)
Comment (myakove) [reply must_gather_explorer/utils.py:285]: Done

File: must_gather_explorer/utils.py [hunk 4]
```diff
@@ -267,71 +272,28 @@ def get_resource_kind_by_alias(resources_aliases: dict[str, list[str]], requeste
...
-        if not resources_raw_data:
-            CONSOLE.print(f"No resources found for {resource_kind} {resource_name} {namespace_name}")
-            return False
-        actions_dict[action_name](resources_raw_data, print_yaml, yaml_fields_to_get)
+        return
 
-    except MissingResourceKindAliasError:
-        return False
-    return False
+    actions_dict[action](resources_raw_data, oyaml, yaml_fields)
```
[the remark this answers is not in the document]
Comment (myakove) [reply must_gather_explorer/utils.py:301]: Actions are managed by `MustGatherExplorerPrompt` and validated there

File: must_gather_explorer/tests/test_cli.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        ("get pod csi- -oyaml"),
+        ("get pod csi-nfs-controller-c54d89cd5-gbd5s -oyaml"),
+        ("get pod -oyaml -f spec.nodeSelector"),
+        ("get pod csi- -oyaml -f spec.nodeSelector"),
+        ("get pod csi-nfs-controller-c54d89cd5-gbd5s -oyaml -f spec.nodeSelector"),
+    ],
+)
+def test_get(test_input, prompt_handler):
+    _, err = run_cmd(prompt_handler, test_input)
+    assert not err, err
```
[the remark this answers is not in the document]
Comment (myakove) [reply must_gather_explorer/tests/test_cli.py:74]: Those tests only cover the CLI args, not code functionality .
Please open issue to add functional tests

**Review by jpeimer — APPROVED** [review]

**Comment by myakove** [comment]
@jpeimer did you check if all working? 
if so please verify and merge :)

**Comment by jpeimer** [comment]
@myakove yes I checked, and it looks like all is working :)

**Pull request merged**